contrib/fwupd.spec.in

```diff
@@ -133,6 +133,8 @@ Obsoletes: fwupd-sign < 0.1.6
 Obsoletes: libebitdo < 0.7.5-3
 Obsoletes: libdfu < 1.0.0
 Obsoletes: fwupd-labels < 1.1.0-1
+Obsoletes: %{name}-plugin-modem-manager < 1.8.6-1
```

Comment:
Why were these split out originally? If it's important to be split out then maybe all plugins except these should be built in.

Reply:
each plugin was 100Kb which was an easy saving on a server that was never going to have either a modem or something supported by flashrom. Now the plugin is 24Kb. If people want to quibble that they don't want libflashrom as a runtime requirement then we can justify that as a reason to port away from libflashrom and copy the tiny bits we do use. I guess it's similar with mm, although we talk to that over DBus so if it's not there it's not a huge problem -- the problem there is the libs like `3.4M	/usr/lib64/libqmi-glib.so.5.8.0`

Reply:
That dependency needed for libqmi and libmbim makes sense to focus on.

Can we load that library only if used? So someone could block the plugin in daemon.conf and not require the dependency?

Reply:
Nah, we get nowhere close to running:

    ./src/fwupdtool: error while loading shared libraries: libqmi-glib.so.5: cannot open shared object file: No such file or directory

Do you think I should do a hybrid solution that allows us to have both modules and built-in plugins? Without the former I'm in trouble with the coreos team too.

Reply:
Yeah my thought is that we should have an option in meson that splits out requested plugins to standalone. It can take a string argument of which plugins should be standalone.

Reply:
I don't think that's possible with this design; the entrypoints are completely different. Will ponder.

Reply:
Another idea - what if we ship optional stubs for those other libraries to use in Fedora?

 If those packages are installed we don't use them. If the stubs are in use "disable" those plugins.

Reply:
Okay, all but two are stand-alone now. 2.2MB lighter!